Plutarch/DataRepr/Internal.hs

```diff
@@ -139,6 +171,10 @@ pdcons = punsafeCoerce $ pcons @PBuiltinList @PData
 pdnil :: Term s (PDataRecord '[])
 pdnil = punsafeCoerce $ pnil @PBuiltinList @PData
 
+-- | Convert a 'PDataRecord' into a builtin list of data values, losing type information in the process.
+pdToBuiltin :: Term s (PDataRecord xs) -> Term s (PBuiltinList PData)
```

Comment:
This is just `pto` AFAICT?

Reply:
Yes, but for some reason I seem to be getting ambiguous type variable on that `b` in `PInner` whenever using `pto` in some of these places: `pto x == pto y`, `pdToBuiltin = pto` doesn't work either here. What am I missing here?

Reply:
something like
```hs
foo :: Term s (PDataRecord l) -> Term s (PBuiltinList PData)
foo x = pto x
```
doesn't work - particularly, it fails with an "type variable b0 is ambiguous" error (the `b0` is the `b` for `PInner`). I think GHC is refusing to expand the `PInner` type family at the presence of that `l` type variable. When `l` is replaced with a concrete type rather than a type variable-
```hs
foo :: Term s (PDataRecord '[]) -> Term s (PBuiltinList PData)
foo x = pto x
```
it works fine.

It looks like `pto` is not going to work with types that take in type variables? Potentially worth making an issue and tackle separately?

Reply:
Yep, this is a bug in the definition of the PlutusType instance. We should fix this if possible.

Reply:
I actually have no idea how, hints?

Reply:
we need to have an instance for all PDataRecords somehow

Reply:
yeah, the wall I've been hitting on that is essentially:

```hs
Couldn't match type ‘labeledX’
               with ‘PLabelOf labeledX ':= PUnLabel labeledX’
```

Minimal repro:

```hs
type PUnLabel :: PLabeledType -> PType
type family PUnLabel a where
  PUnLabel (name ':= a) = a

type PLabelOf :: PLabeledType -> Symbol
type family PLabelOf a where
  PLabelOf (name ':= a) = name
```

GHC doesn't seem to realize `labeledX` will always be equal to `PLabelOf labeledX ':= PUnLabel labeledX` - is there a way to assert that? (other than having it as a constraint in the instance head)

Reply:
This constraint meets your stated requirements:

    instance (labeledX ~ PLabelOf labeledX ':= PUnLabel labeledX) => Thingy (labeledX ': xs) where

Reply:
yeah but I believe I need this to work for all `x ': xs`, or even all `xs` (both `'[]` and `x ': xs`), without any constraints in the instance head. The constraint will need to be propagated all the way and it just seems a bit unergonomic.